xonsh/tools.py

```diff
@@ -427,9 +427,20 @@ def print_exception(msg=None):
     env = getattr(builtins, '__xonsh_env__', os.environ)
     if 'XONSH_SHOW_TRACEBACK' not in env:
         sys.stderr.write('xonsh: For full traceback set: '
-                         '$XONSH_SHOW_TRACEBACK = True\n')
-    if env.get('XONSH_SHOW_TRACEBACK', False):
-        traceback.print_exc()
+                         '$XONSH_SHOW_TRACEBACK = True or '
+                         '$XONSH_SHOW_TRACEBACK = \'<filename>\'\n')
+    # get env option for traceback and convert it to a tuple
+    traceback_opt = env.get('XONSH_SHOW_TRACEBACK', False)
+    show_trace, out_file = to_tb_tuple(traceback_opt)
+    if show_trace:
+        if not out_file:
+            # if the user has not specified a file, output to stderr
+            traceback.print_exc()
+        else:
+            # if the user has specified a file, open it in append mode
+            # and output traceback info there, as well as to stderr
+            traceback.print_exc()
```

Comment:
I also think that if a user has specified a file as the argument, it probably means they don't want to see the whole traceback in the shell.  We could try to find a way to toggle between these behaviors, but I think it would make sense to display only the error message here, and write the full traceback to the specified file.